src/pretix/api/serializers/event.py

```diff
@@ -202,6 +202,105 @@ class Meta:
                   'presale_start', 'presale_end', 'location', 'event',
                   'item_price_overrides', 'variation_price_overrides', 'meta_data')
 
+    def validate(self, data):
+        data = super().validate(data)
+        event = self.context['request'].event
+
+        full_data = self.to_internal_value(self.to_representation(self.instance)) if self.instance else {}
+        full_data.update(data)
+
+        Event.clean_dates(data.get('date_from'), data.get('date_to'))
+        Event.clean_presale(data.get('presale_start'), data.get('presale_end'))
+
+        SubEvent.clean_items(event, [item['item'] for item in full_data.get('subeventitem_set')])
```

Comment:
I haven't tested it, but I have the feeling that ``validate`` runs before the serializer-level variation, so this might throw a KeyError in case anyone submits a ``subevent_item`` without a ``item`` key, leading to a ``500`` instead of ``400``.

Reply:
Do you have a proposal how to handle this?

I could probably just extend it to 
```[item['item'] if 'item' in item else None for item in full_data.get('subeventitem_set')]``` and have `clean_items()` skip those entries... But if there is a better way, I'd like to learn it :)

Reply:
Nevermind, I actually worked around this by filtering incomplete objects in a new `validate_item_price_overrides`.